xonsh/lexer.py

```diff
@@ -33,53 +327,26 @@ def __init__(self, errfunc=lambda e, l, c: print(e)):
         self.errfunc = errfunc
         self.fname = ''
         self.last = None
-        self.lexer = None
-        self.indent = ''
-        self.in_py_mode = [True]
 
     def build(self, **kwargs):
         """Part of the PLY lexer API."""
-        self.lexer = lex.lex(object=self, **kwargs)
-        self.reset()
+        pass
 
     def reset(self):
-        self.lexer.lineno = 1
-        self.indent = ''
-        self.last = None
-        self.in_py_mode = [True]
-        self.in_parens = [False]
-
-    @property
-    def lineno(self):
-        if self.lexer is not None:
-            return self.lexer.lineno
-
-    @lineno.setter
-    def lineno(self, value):
-        if self.lexer is not None:
-            self.lexer.lineno = value
+        pass
 
     def input(self, s):
         """Calls the lexer on the string s."""
-        self.lexer.input(s)
+        s = re.sub(r'#.*', '', s)
+        self.token_stream = preprocess_tokens(tok(s))
 
     def token(self):
         """Retrieves the next token."""
-        self.last = self.lexer.token()
-        return self.last
-
-    def token_col(self, token):
-        """Discovers the token column number."""
-        offset = self.lexer.lexdata.rfind('\n', 0, token.lexpos)
-        return token.lexpos - offset
-
-    def _error(self, msg, token):
-        location = self._make_tok_location(token)
-        self.errfunc(msg, location[0], location[1])
-        self.lexer.skip(1)
-
-    def _make_tok_location(self, token):
-        return (token.lineno, self.token_col(token))
+        try:
+            self.last = next(self.token_stream)
+            return self.last
+        except StopIteration:
+            return None
```

Comment:
I did it this way because I always wanted `self.last` to be a valid token once we started going.  Looking at the parser again, though, it seems like that might not be a concern.